service/src/main/java/uk/gov/hmcts/reform/fpl/service/payment/FeeService.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.fpl.service.payment;
+
+import com.google.common.collect.ImmutableList;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.reform.fpl.config.payment.FeesConfig;
+import uk.gov.hmcts.reform.fpl.enums.payment.FeeType;
+import uk.gov.hmcts.reform.fpl.model.payment.fee.FeeParameters;
+import uk.gov.hmcts.reform.fpl.model.payment.fee.FeeResponse;
+
+import java.net.URI;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static org.springframework.web.util.UriComponentsBuilder.fromHttpUrl;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class FeeService {
+
+    private final RestTemplate restTemplate;
```

Comment:
Couldn't you use @Feign here? the url building seems a bit ugly, please check OrganisationApi. Don't mind if it works but I think Feign is a bit nicer (but there are ea bit issues with conf)